src/TestFramework/PhpUnit/Adapter/PhpUnitAdapter.php

```diff
@@ -84,14 +83,6 @@ public function getMemoryUsed(string $output): float
         return -1;
     }
 
-    public function getMutationsCollectionNodeVisitors(): array
-    {
-        return [
-            100 => new PhpUnitClassCodeCoverageIgnoreVisitor(),
-            15 => new PhpUnitMethodCodeCoverageIgnoreVisitor(),
```

Comment:
Previously these 2 visitors where executed in a completely different places

```php

// PhpUnitClassCodeCoverageIgnoreVisitor <------ here

$traverser->addPrioritizedVisitor(new NotMutableIgnoreVisitor(), 50);
$traverser->addPrioritizedVisitor(new ParentConnectorVisitor(), 40);
$traverser->addPrioritizedVisitor(new FullyQualifiedClassNameVisitor(), 30);
$traverser->addPrioritizedVisitor(new ReflectionVisitor(), 20);

// PhpUnitMethodCodeCoverageIgnoreVisitor <--- here
```

now, they both executed in one place. Did you manage to understand what was the reason for these 2 visitors be on the different places? And why doesn't it affect anything anymore?

Reply:
In the original implementation we used reflection to get the doc comment on the `PhpUnitMethodCodeCoverageIgnoreVisitor`, thus we needed to add it after the `ReflectionVisitor`. Later this was changed to be more in line with the Class version and just use the AST information of the doc comment, but the priority wasn't changed to the top.

Running it as soon as possible is a micro optimization as we traverse less nodes. But it should at the very least be ran before the mutation collection, as it prevents useless mutations (signature of interfaces/abstract methods), and ignored mutations (with the code coverage ignore annotation)